suite-common/wallet-core/src/accounts/accountsActions.ts

```diff
@@ -0,0 +1,135 @@
+import { createAction } from '@reduxjs/toolkit';
+
+import { AccountInfo } from '@trezor/connect';
+import { Account, SelectedAccountStatus, DiscoveryItem } from '@suite-common/wallet-types';
+import {
+    enhanceAddresses,
+    enhanceTokens,
+    enhanceUtxo,
+    formatNetworkAmount,
+    getAccountKey,
+    getAccountSpecific,
+} from '@suite-common/wallet-utils';
+
+import { actionPrefix } from './constants';
+
+const disposeAccount = createAction(`${actionPrefix}/disposeAccount`);
+
+const updateSelectedAccount = createAction(
+    `${actionPrefix}/updateSelectedAccount`,
+    (payload: SelectedAccountStatus): { payload: SelectedAccountStatus } => ({
+        payload,
+    }),
+);
+
+const removeAccount = createAction(
```

Comment:
because sometimes you may need to prepare actions that won't have only payloads. Like this `{ type: '...', payload: 'aaa', somethingElse: 'aaa', 'meta': 'bbb' }` etc. We even do this in suite, but I believe it was done by mistake and it most of cases you should place everything to payload. 

This is just how createAction works, you can study it in RTK docs, https://redux-toolkit.js.org/api/createAction as you can see second param is called prepareAction (not preparePayload).

Also we can't use generic for payload type because:
```
const action = createAction<Account[]>(`${actionPrefix}/removeAccount`)
// will generate this type =>
type Action = {
	type: string,
    payload: Account[]
}
```

As you can see that `type: string` is not correct because we need type where exact action type like is defined not as generic strict, because it will break that Action discriminited union which we need to keep in Suite for compatibility issue. When you use only RTK action matchers without any these kind of unions it is not issue at all.

I had discussion about this with RTK maintainers: https://github.com/reduxjs/redux-toolkit/issues/2614#issuecomment-1220744758

Sadly we can't use also slices because of cyclic dependencies (which are much worse that cyclic imports, which could be easily solved by `import type`)

Reply:
Ok, meaning it's like that and not like:

`const removeAccount = createAction<Account[]>('${actionPrefix}/removeAccount');`

because you didn't want the type to be `string`, got it.